legend-engine-xt-serviceStore-grammar/src/main/java/org/finos/legend/engine/language/pure/compiler/toPureGraph/ServiceStoreCompilerExtension.java

```diff
@@ -102,16 +105,42 @@ public List<Function2<Connection, CompileContext, Root_meta_pure_runtime_Connect
                         ServiceStoreConnection serviceStoreConnection = (ServiceStoreConnection) connectionValue;
 
                         Root_meta_external_store_service_metamodel_runtime_ServiceStoreConnection pureServiceStoreConnection = new Root_meta_external_store_service_metamodel_runtime_ServiceStoreConnection_Impl("", null, context.pureModel.getClass("meta::external::store::service::metamodel::runtime::ServiceStoreConnection"));
-                        pureServiceStoreConnection._element(HelperServiceStoreBuilder.getServiceStore(serviceStoreConnection.element, serviceStoreConnection.elementSourceInformation, context));
+                        Root_meta_external_store_service_metamodel_ServiceStore pureServiceStore = HelperServiceStoreBuilder.getServiceStore(serviceStoreConnection.element, serviceStoreConnection.elementSourceInformation, context);
+                        pureServiceStoreConnection._element(pureServiceStore);
                         pureServiceStoreConnection._baseUrl(serviceStoreConnection.baseUrl);
-                        HelperServiceStoreBuilder.compileAndAddAuthenticationSpecifications(pureServiceStoreConnection,serviceStoreConnection.authenticationSpecifications,context);
+                        HelperServiceStoreBuilder.compileAndAddAuthenticationSpecifications(pureServiceStoreConnection, serviceStoreConnection.authenticationSpecifications, pureServiceStore._securitySchemes().getMap(), context);
                         return pureServiceStoreConnection;
                     }
                     return null;
                 }
         );
     }
 
+    @Override
+    public List<Function2<SecurityScheme, CompileContext, Root_meta_external_store_service_metamodel_SecurityScheme>> getExtraSecurityProcessors()
+    {
+        return Lists.mutable.with((scheme, context) ->
+        {
+            if (scheme instanceof HttpSecurityScheme)
+            {
+                HttpSecurityScheme httpSecurityScheme = (HttpSecurityScheme) scheme;
+                return new Root_meta_external_store_service_metamodel_HttpSecurityScheme_Impl("", null, context.pureModel.getClass("meta::external::store::service::metamodel::HttpSecurityScheme"))
+                        ._scheme(context.resolveEnumValue("meta::external::store::service::metamodel::Scheme", httpSecurityScheme.scheme.name()))
+                        ._bearerFormat(httpSecurityScheme.bearerFormat);
```

Comment:
Does it make sense to add a check that bearerFormat should not be specified with basic scheme. If yes add a compiler test for this too